File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int, int> map;
        vector<int> ans;
        for(int i=0;i<n;i++){
            int first=nums[i];
            int second=target-first;
            if(map.find(second)!=map.end()){
                ans.push_back(i);
                ans.push_back(map[second]);
                break;
            }
            map[first]=i;
        }
    return ans;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        unique = {}
        for i in range (0,len(nums)):
            unique[nums[i]] = i

        for i in range(0,len(nums)):
            if target-nums[i] in unique and i!= unique[target-nums[i]]:
                return i, unique[target-nums[i]]
        



        #BruteForce
        '''l = len(nums)
        print(l)
        for i in range(0,l):
            for j in range(i+1,l):
                if nums[i] + nums[j] == target:
                    return i,j '''

